### PR TITLE
EVAKA-4203: prevent changes to vasu template content once it is in use

### PR DESCRIPTION
**Pull request by Joosakur**
#### Summary
readonly mode in frontend for used vasu templates

### CODE REVIEW

File: frontend/src/employee-frontend/components/vasu/templates/VasuTemplateEditor.tsx [hunk 1]
```diff
@@ -423,6 +435,7 @@ export default React.memo(function VasuTemplateEditor() {
                               }
                               text={i18n.vasuTemplates.addNewQuestion}
                               icon={faPlus}
+                              disabled={readonly}
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/vasu/templates/VasuTemplateEditor.tsx:438]: Would it make more sense to drop these buttons completely if the template is readonly?

**Review by jaahon — COMMENTED** [review]
I think backend should check that there aren't any documents created off the template when the edit is saved.

File: frontend/src/employee-frontend/components/vasu/templates/VasuTemplateEditor.tsx (hunk 1 above)
Comment (Joosakur) [reply frontend/src/employee-frontend/components/vasu/templates/VasuTemplateEditor.tsx:438]: https://github.com/espoon-voltti/evaka/blob/60f8035d0c6ec607f6465ad807a8e3bdb577a648/service/src/main/kotlin/fi/espoo/evaka/vasu/VasuTemplateController.kt#L149 backend validation actually is there already. You were probably looking at the other PUT endpoint which updates e.g. template date range and allows limited edits even after template is in use.

Comment (jaahon) [reply frontend/src/employee-frontend/components/vasu/templates/VasuTemplateEditor.tsx:438]: Sorry, my bad.

**Review by jaahon — APPROVED** [review]

**Pull request merged**